web_poet/page_inputs/http.py

```diff
@@ -18,12 +19,46 @@
 _AnyStrDict = Dict[AnyStr, Union[AnyStr, List[AnyStr], Tuple[AnyStr, ...]]]
 
 
-class ResponseURL(str):
+class _URL:
```

Comment:
We could move this later to `w3lib` when we're comfortable in the "smart" URL functionalities that we like.